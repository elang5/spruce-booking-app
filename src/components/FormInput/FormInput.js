import React from 'react'

export default function FormInput({ id, field, handleFields, name, type }) {
  return (
    <div className='input-cell'>
      <label htmlFor={id}>
        {field}
      </label>
      <input 
        required
        type={type}
        onChange={handleFields}
        id={id.toLowerCase().replace(/\s/g, '')}
        name={name}/>
    </div>
  )
}

FormInput.defaultProps = {
  id: '',
  field: '',
  type: '',
  name: ''
}