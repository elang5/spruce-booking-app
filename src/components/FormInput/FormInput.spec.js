import React from "react";
import ReactDOM from "react-dom";
import FormInput from "./FormInput";
import renderer from "react-test-renderer";

describe("FormInput Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(< FormInput />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(<FormInput/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
