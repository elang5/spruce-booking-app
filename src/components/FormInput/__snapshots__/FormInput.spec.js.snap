// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component renders the UI as expected 1`] = `
<div
  className="input-cell"
>
  <label
    htmlFor=""
  >
    
  </label>
  <input
    id=""
    name=""
    required={true}
    type=""
  />
</div>
`;
